src/GraphQL/Types/Scalars/IdGraphType.cs

```diff
@@ -23,31 +23,39 @@ public IdGraphType()
             //    "as `\"4\"`) or integer (such as `4`) input value will be accepted as an `ID`.";
         }
 
-        /// <inheritdoc/>
-        public override object Serialize(object value) => value.ToString();
-
         /// <inheritdoc/>
         public override object ParseLiteral(IValue value) => value switch
         {
             StringValue str => ParseValue(str.Value),
             IntValue num => num.Value,
             LongValue longVal => longVal.Value,
+            BigIntValue bigIntValue => bigIntValue.Value,
             _ => null,
         };
 
-        /// <inheritdoc/>
-        public override object ParseValue(object value) => value.ToString().Trim(' ', '"');
-
         /// <inheritdoc/>
         public override bool CanParseLiteral(IValue value) => value switch
         {
             StringValue _ => true,
             IntValue _ => true,
             LongValue _ => true,
+            BigIntValue _ => true,
+            NullValue _ => true,
             _ => false
         };
 
         /// <inheritdoc/>
-        public override IValue ToAST(object value) => new StringValue(value.ToString());
+        public override object ParseValue(object value) => value switch
+        {
+            int _ => value,
+            long _ => value,
+            _ => value?.ToString()
+        };
+
+        /// <inheritdoc/>
+        public override bool CanParseValue(object value) => true;
+
+        /// <inheritdoc/>
+        public override object Serialize(object value) => value?.ToString();
```

Comment:
```suggestion
        public override object Serialize(object value) => ParseValue(value)?.ToString();
```

Reply:
Serialize should handle Guids.  They are commonly used for identifiers.  Normally I would say that ParseValue should not interpret them, as external representations should be strings (ideally), or numbers.  I can put it in there anyway - ?

Reply:
> Serialize should handle Guids.

OK but not bools anyway.

> Normally I would say that ParseValue should not interpret them

🤔 OK. So `ParseValue` and `Serialize` will be different a little.

Reply:
I didn't separate/duplicate the code, so as it is now, since Serialize calls ParseValue, ParseValue can handle a Guid.... lmk if you think it should be changed.

Reply:
OK. I will review again.